csrc/multidevice/communication.h

```diff
@@ -15,43 +15,39 @@
 
 namespace nvfuser {
 
-/*
-  This struct gathers all the parameters necessary for the
-  construction a communication
-*/
+// This struct gathers all the parameters necessary for the
+// construction a communication
 struct CommParams {
   DeviceIdxType root = -1;
-  std::vector<at::Tensor> src_bufs;
-  std::vector<at::Tensor> dst_bufs;
-  Team team; // should not have duplicate
+  bool is_root_in_mesh = true;
+  Team team; // should not have duplicates and should contain both the root and
+             // the mesh
   c10d::ReduceOp::RedOpType redOp = c10d::ReduceOp::RedOpType::UNUSED;
+  int64_t scattered_axis = -1;
 };
 
-/*
-The class "Communication" represents a MPI-style communication
-communication operation to be executed on the network. The base class
-Communication should not be used directly but through its derived classes:
-Broadcast, Gather, Scatter, Allgather, and SendRecv. Other collectives will be
-added later.
-
-CommParams contains the arguments for the communication constructors.
-Note that each process (associated with a device index given by
-communicator.deviceId()) will fill CommParams with different arguments,
-depending on the role they play in this communication. For example, the root of
-a Gather communication will have <team_size> destination buffers, whereas
-non-root will have no destination buffers. Also, the ranks not participating in
-the communication should not instantiate it.
-
-The method "post" triggers the execution of the communication. This call is
-non-blocking. The communication can be posted multiple times.
-It is assumed that the current device_index (given by
-communicator.deviceId()) belongs to the team of the communication,
-otherwise an error is thrown.
-
-NOTE: pytorch's NCCL process group API needs <team_size> buffers on root for
-scatter/gather operation.
-*/
-
+// The class "Communication" represents a MPI-style communication
```

Comment:
what motivates this change? The multiline comment is nicer imo, and used in other places of the codebase, e.g., https://github.com/NVIDIA/Fuser/blob/d12b786dd7782c0ea5d087c36fd0497ad92ff7dd/csrc/ir/base_nodes.h#L31

Also, did you change the content or only replace the multi-line comments with the inline comments?

Reply:
To make style consistent in one codebase. Yes, I'm sure `/* */` is used in some other files but they appear much less often than `//` when I search both in our codebase. Do you have a technical problem (e.g. IDE couldn't line wrap correctly) with `//`? 

> Also, did you change the content or only replace the multi-line comments with the inline comments?

No. Sorry, I should have called that out to save you some review time.